lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -549,6 +552,82 @@ def upload_csv_to_report_store(rows, csv_name, course_id, timestamp):
     )
 
 
+def is_eligible_for_certificate(embargoed, whitelisted, grade):
+    """
+    Check if user is eligible for certificate.
+    """
+    return not embargoed and (whitelisted or grade is not None)
+
+
+def is_certificate_delivered(embargoed, whitelisted, certificate_status):
+    """
+    Check if certificate is generated and delivered to user.
+    """
+    return not embargoed and (whitelisted or certificate_status['status'] == CertificateStatuses.downloadable)
+
+
+def verification_status(user, course_id):
+    """
+    Returns the verification status.
+    """
+    user_is_verified = SoftwareSecurePhotoVerification.user_is_verified(user)
+    user_is_re_verified = SoftwareSecurePhotoVerification.user_is_reverified_for_all(course_id, user)
+
+    if not user_is_re_verified:
+        return 'ID Verification Expired'
+
+    if user_is_verified and user_is_re_verified:
+        return 'ID Verified'
+    else:
+        return 'Not ID Verified'
+
+
+def certificate_type(certificate_status):
+    """
+    Returns certificate mode if status is not 'unavailable'
+    """
+    if certificate_status['status'] == CertificateStatuses.unavailable:
+        return 'N/A'
+    return certificate_status['mode']
+
+
+def generate_certificates_data(user, course_id, grade):
+    """
+    Generates and return the user data related to the certificates for a particular course.
+    """
+
+    user_in_white_list = CertificateWhitelist.objects.filter(user=user, course_id=course_id, whitelist=True).exists()
```

Comment:
This list is going to be very small and so to prevent a query for every single user, it would be simpler to fetch it once in the calling function and then just do the check in Python.